pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -89,7 +91,8 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env model.Environmen
 
 			// create default cluster
 			clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port)
-			defaultCluster := buildDefaultCluster(env, clusterName, convertResolution(service.Resolution), hosts)
+			serviceAccounts := env.ServiceAccounts.GetIstioServiceAccounts(service.Hostname, []string{port.Name})
+			defaultCluster := buildDefaultCluster(env, clusterName, convertResolution(service.Resolution), hosts, serviceAccounts)
```

Comment:
Alternate simpler implementation suggestion: when a destinationrule is obtained, pass it to a convertIstioMtlsToMtls function. This function will simply set the tls mode to mutual with the cert file names using istio certs and SAN field using the service accounts. Once converted, the rest of the code does not need to be changed at all. And if config is nil, create a new dest rule if mesh.mtls is true and set the mTLS settings in the dest rule. This way you don’t have to wire up tnt service accounts everywhere. 
In future, if anyone were to add more cluster building code, they wouldn’t have to worry about plumbing service accounts everywhere. If you want, you could add this code directly into model.DestinationRules() such that you don’t even have to change cluster.go

Reply:
I suggest the conversion as it ensures that any other code using cluster.go functions does not end up overriding auth stuff.

Reply:
this conversion is better done in cluster.go, instead of model code due to certain wildcard issues in destination rules. But please convert so that you don’t have to push the service account thing into all functions.

Reply:
It's a good suggestion. In fact, it's similar to authn policy: when there is no authn policy in CRD, we generate one based on config map settings. May be we can do the same here?